framework/src/meshgenerators/ParsedGenerateNodeset.C

```diff
@@ -39,13 +39,20 @@ ParsedGenerateNodeset::validParams()
       {},
       "Vector of values for the constants in constant_names (can be an FParser expression)");
 
+  params.addParam<bool>(
+      "allow_distributed_meshes",
+      false,
+      "Can be used to avoid erroring on distributed meshes, when it is known to be safe. For "
+      "example, when only using a parsed expression to select nodes.");
```

Comment:
Should we be laying this on the user?  I'm not sure what combinations of options are known to be safe, and I'm looking at the source code right now!

Reply:
the parent classes options could not be safe

for example if we dont know the niehgbor subdomains because we dont know the neighbor

Reply:
well we dont offer included_neighbors for nodesets.
Still I have not vetted or tested that code for distributed mesh

Reply:
Ah, I missed that we're inheriting `excluded_subdomains` and `included_subdomains`.

Would it make sense to handle disabling errors based on the current `NodeSetsGeneratorBase` options?  Or are you worried someone will add new parent options in the future that will obviate our child handling?

Reply:
raised #31583